### PR TITLE
Add doc for nn.MultiMarginLoss (shape, example)

### LINKED ISSUE #63747: torch.nn.MultiMarginLoss does not have doc for shape
## 📚 Documentation

This is the doc with missing documentation for input/output shape:
https://pytorch.org/docs/stable/generated/torch.nn.MultiMarginLoss.html

This is another similar class, which has documentation for input/output shape:
https://pytorch.org/docs/stable/generated/torch.nn.MultiLabelMarginLoss.html

### PR DESCRIPTION
**Pull request by changjonathanc**
Fixes #63747

### CODE REVIEW

File: torch/nn/modules/loss.py [hunk 1]
```diff
@@ -1346,6 +1346,20 @@ class MultiMarginLoss(_WeightedLoss):
...
+    Shape:
+        - Input: :math:`(N, C)` where `N` is the batch size and `C` is the number of classes.
+        - Target: :math:`(N)` where each value is :math:`0 \leq \text{targets}[i] \leq C-1`.
+        - Output: scalar. If :attr:`reduction` is ``'none'``, then :math:`(N)`.
+
+    Examples::
+
+        >>> loss = nn.MultiMarginLoss()
+        >>> x = torch.FloatTensor([[0.1, 0.2, 0.4, 0.8]])
+        >>> y = torch.LongTensor([3])
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:1359]: nit: FloatTensor and LongTensor are the old deprecated ways of creating tensors; `torch.tensor` is preferred now

File: torch/nn/modules/loss.py [hunk 2]
```diff
@@ -1346,6 +1346,20 @@ class MultiMarginLoss(_WeightedLoss):
             elements in the output, ``'sum'``: the output will be summed. Note: :attr:`size_average`
             and :attr:`reduce` are in the process of being deprecated, and in the meantime,
             specifying either of those two args will override :attr:`reduction`. Default: ``'mean'``
+
+    Shape:
+        - Input: :math:`(N, C)` where `N` is the batch size and `C` is the number of classes.
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:1351]: formatting nit: \`N\` -> :math:\`N\` and \`C\` -> :math:\`C\` makes this look a bit nicer

Comment (jbschlosser) [reply torch/nn/modules/loss.py:1351]: Also, since the input allows for no batch dim, this should read:

```
Input: :math:`(N, C)` or :math:`(C)`, where :math:`N` is the batch size and :math:`C` is the number of classes.
```

File: torch/nn/modules/loss.py [hunk 3]
```diff
@@ -1346,6 +1346,20 @@ class MultiMarginLoss(_WeightedLoss):
...
+        - Target: :math:`(N)` where each value is :math:`0 \leq \text{targets}[i] \leq C-1`.
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:1352]: target allows for no batch dim:

```
Target: :math:`(N)` or :math:`()`, where each value is...
```

**Review by jbschlosser — COMMENTED** [review]
Thanks for the contribution! Looks great - just a couple minor comments below

File: torch/nn/modules/loss.py (hunk 1 above)
Comment (cccntu) [reply torch/nn/modules/loss.py:1359]: Did not know about this, I just copied from another example. Thanks!

File: torch/nn/modules/loss.py [hunk 4]
```diff
@@ -1346,6 +1346,20 @@ class MultiMarginLoss(_WeightedLoss):
             elements in the output, ``'sum'``: the output will be summed. Note: :attr:`size_average`
             and :attr:`reduce` are in the process of being deprecated, and in the meantime,
             specifying either of those two args will override :attr:`reduction`. Default: ``'mean'``
+
+    Shape:
+        - Input: :math:`(N, C)` or :math:`(C)`, where :math:`N` is the batch size and :math:`C` is the number of classes.
+        - Target: :math:`(N)` or :math:`()`, where each value is :math:`0 \leq \text{targets}[i] \leq C-1`.
+        - Output: scalar. If :attr:`reduction` is ``'none'``, then :math:`(N)`.
```
Comment (jbschlosser) [inline torch/nn/modules/loss.py:1353]: nit: now that the target shape has been updated, I think this should read something like:
```
If :attr:`reduction` is ``'none'``, then same shape as the target
```

**Review by jbschlosser — COMMENTED** [review]
Changes look good! Just one tiny thing left and good to go

**Review by jbschlosser — APPROVED** [review]
LGTM!
